app/models/event.rb

```diff
@@ -3,6 +3,24 @@ class Event < ApplicationRecord
   validates_presence_of :description
   validates_presence_of :eventType
   has_many :event_statuses
-  has_many :users, :through => :event_statuses
+  has_many :users, :through => :event_statuses, validate: false
+  validate :check_users
+  validate :check_preferences
   enum eventType: %i[event info message]
+
+  private
+  def check_users
+    if self.users.blank?
+      errors.add(:users, I18n.t('global.error_users'))
+    end
+  end
+
+  def check_preferences
+    if !self.use_email? && !self.use_call? && !self.use_text?  && !self.use_app?
```

Comment:
What is `self.___` referring to here? `self.use_email` is not an attribute for the event model? Why is an event responsible for contact preferences? A user should dictate their own preferences, so this should probably be in that model?